exp-player/addon/components/exp-player.js

```diff
@@ -1,59 +1,40 @@
 import Ember from 'ember';
 import layout from '../templates/components/exp-player';
+import parseExperiment from '../utils/parse-experiment';
 
 export default Ember.Component.extend({
     layout: layout,
-    store: Ember.inject.service('store'),
 
+    experiment: null, // Experiment model
     frames: null,
-    frameIndex: null,
-    _last: null,
-    ctx: {
-        data: {}
+
+    frameIndex: null,  // Index of the currently active frame
+
+    expData: {},  // Temporarily store data collected until we sent to server at end
+
+    init: function() {
+        this._super(...arguments);
+        this.set('frameIndex', 0);
+        var frameConfigs = parseExperiment(this.get('experiment.structure'));
```

Comment:
Will this allow us to easily capture the generated parameters (for random frames, etc)?